### PR TITLE
Feature - introduce markdown tables

### PR DESCRIPTION
**Pull request by anusaini**
Introduced markdown tables.
Return value is now an object { ascii, markdown }
Version bump.

### CODE REVIEW

**Comment by gabrielcsapo** [comment]
This is awesome! What if instead of overloading the return value have something like

```js
Table({}).markdown() 
```

The default will be just an ascii value for the `toString`

File: index.js [hunk 1]
```diff
@@ -6,12 +6,16 @@
...
+ * @return {Object} - { ascii - a string that represents the ascii table of the data provided,
+ *                      html - a provided html version of table,
+ *                      table - html <table> table,
+ *                      markdown - a markdown table }
  */
 module.exports = function turtler(data, options={}) {
   if(!Array.isArray(data)) throw new Error('data should be an array of arrays');
 
-  let table = '';
+  let asciiTable = '';
```
Comment (gabrielcsapo) [inline index.js:17]: Yeah I am sure that this is a good addition but should be retrieved using a getter method.

for example

```
get ascii () {
 return this.toString();
}
get markdown() {
 return // code to make it a marko
}
```

File: index.js [hunk 2]
```diff
@@ -6,12 +6,16 @@
  * @param  {Boolean} options.hasHeader - this will determine if the table data has a header or not, by default this is true
  * @param  {String}  options.columnSeparator - the default separators is ' | '
  * @param  {String}  options.headerSeparator - the default header separators is '=', this value should only be one character
- * @return {String} - a string that represents the ascii table of the data provided
+ * @return {Object} - { ascii - a string that represents the ascii table of the data provided,
+ *                      html - a provided html version of table,
+ *                      table - html <table> table,
+ *                      markdown - a markdown table }
  */
 module.exports = function turtler(data, options={}) {
```
Comment (gabrielcsapo) [inline index.js:14]: what we should do is make the main file return a turtler class, so that we can extend the object.

**Review by gabrielcsapo — CHANGES_REQUESTED** [review]

**Comment by anusaini** [comment]
I agree; a class with appropriate methods makes much more sense. And default toString can just return ascii table. Let me find some time; today feels like Monday.

**Comment by gabrielcsapo** [comment]
I have something ready let me submit a PR and you can review it. I will add you as the coauthor

**Comment by gabrielcsapo** [comment]
going to close this as I have opened another PR with the same content. https://github.com/gabrielcsapo/turtler/pull/5 Please review here

**Pull request closed without merging**